xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-ui/src/main/resources/XWiki/DeletedDocuments.xml

```diff
@@ -343,76 +369,65 @@ XWiki.index.trash.documents.confirmDelete = function(event, table, rowIdx) {
+  deleteDocumentConfirmModal.find('.btn-danger').on('click', function() {
+    var notification = new XWiki.widgets.Notification(
+      $jsontool.serialize($services.localization.render('platform.index.trashDocumentsDeleteInProgress')),
+      'inprogress'
+    );
+    $.post(XWiki.currentDocument.getURL('delete'), {
+        'confirm': 1,
+        'emptybin': true,
+        'form_token': xm.form_token,
+        'ajax': 1
+    }).done(function() {
+      notification.replace(new XWiki.widgets.Notification(
+        $jsontool.serialize($services.localization.render('platform.index.trashDocumentsDeleteDone')),
+        'done'
+      ));
+      trashDocumentsLivetable.refresh();
+    }).fail(function() {
+      notification.replace(new XWiki.widgets.Notification(
+        $jsontool.serialize($services.localization.render('platform.index.trashDocumentsDeleteFailed')),
+        'error'
+      ));
+    });
+  });
+  var createTrashAllButton = function() {
+    if ($('#trashAllDeleted').length == 0) {
+      var actionAllContainer = $('#documentsTrash td.xwiki-livetable-display-header-filter:last-child');
+      var trashAll = $('&lt;a&gt;').attr({
+        'id': 'trashAllDeleted',
+        'class': 'btn btn-danger',
+        'href':'#',
+        'data-toggle': 'modal',
+        'data-target': '#deleteDocumentConfirmModal'
+      }).text("$services.localization.render('index.deletedDocuments.deleteAll')");
+      trashAll.appendTo(actionAllContainer);
+    }
+    $('#trashAllDeleted').toggleClass('disabled', trashDocumentsLivetable.totalRows &lt;= 0);
+  };
 
-// we need to wait the complete display of the livetable to get the number of rows
-document.observe('xwiki:livetable:documentsTrash:displayComplete', function() {
-  // avoid to get multiple times the button
-  if (!$('trashAllDeleted')) {
-    var actionAllContainer = $('documentsTrash').
-      getElementsBySelector('td.xwiki-livetable-display-header-filter:last-child')[0];
+  $(document).on('xwiki:livetable:documentsTrash:displayComplete', createTrashAllButton);
 
-    var trashAll = new Element('a', {
-      'id': 'trashAllDeleted',
-      'class': 'btn btn-danger',
-      'href':'#'
-    }).update("$services.localization.render('index.deletedDocuments.deleteAll')");
-    actionAllContainer.insert(trashAll);
-    trashAll.observe('click', XWiki.index.trash.documents.confirmDeleteAll.bindAsEventListener(trashAll));
-  }
+  // we call the method in case the livetable has already been loaded and the event triggered.
+  createTrashAllButton();
+});
 
-  $('trashAllDeleted').toggleClassName('disabled', trashDocumentsLivetable.totalRows &lt;= 0);
+/*
+* The livetable is hidden when javascript is disabled, so that an alternative &lt;noscript&gt; table is displayed.
+* Re-display it when the document loads and Javascript is available.
+*/
+document.observe('xwiki:dom:loaded', function () {
+  if ($('documentsTrash')) {
+      $('documentsTrash').up('.hidden').removeClassName('hidden');
```

Comment:
I think we can move this inside the Require.js block above (converting the code to jQuery).

Reply:
Actually I tried that, and it wasn't working at all with a `$(document).on('wiki:dom:loaded', ...)`. When I looked on the doc and elsewhere in the wiki it seems this one is always called with a `document.observe`...